crates/polars-core/src/series/mod.rs

```diff
@@ -304,12 +304,15 @@ impl Series {
         if !dtype.is_known() || (dtype.is_primitive() && dtype == self.dtype()) {
             return Ok(self.clone());
         }
-        let ret = self.0.cast(dtype);
```

Comment:
Minor performance improvement here, operation is wasted if the null path is taken.